### PR TITLE
tests: add support for debian 12 and remove support for debian 10

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Update tests to support debian 12.
Also changes from snapd-testing-tools have been imported to support new systems.

### CODE REVIEW

File: tests/lib/external/snapd-testing-tools/utils/spread-shellcheck [hunk 1]
```diff
@@ -124,11 +124,16 @@ def checksection(data, env: Dict[str, str]):
         export_disabled_warnings = set()
         def replacement(match):
             if match.group(0) == '"':
+                # SC2089 and SC2090 are about quotes vs arrays
```
Comment (MiguelPires) [inline tests/lib/external/snapd-testing-tools/utils/spread-shellcheck:127]: Thanks for adding these comments

File: tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml [hunk 2]
```diff
@@ -145,16 +151,16 @@ execute: |
...
+            os.query is-opensuse 15.4
+            ! os.query is-opensuse tumbleweed
             os.query is-classic
             ! os.query is-core
             ;;
-        opensuse-15.4-64)
+        opensuse-15.5-64)
             os.query is-opensuse
-            os.query is-opensuse 15.4
-            ! os.query is-opensuse tumbleweed
```
Comment (MiguelPires) [inline tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml:157]: Was removing this assertion intentional?

**Review by MiguelPires — APPROVED** [review]
LGTM, thanks

File: tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml:157]: yes

Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml:157]: for opensuse-15.5-64) now it is checking os.query is-opensuse 15.5 
and for opensuse-15.4-64) we check ! os.query is-opensuse tumbleweed

**Pull request merged**